packages/overlay/src/App.tsx

```diff
@@ -1,8 +1,11 @@
-import { useEffect, useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import Debugger from './components/Debugger';
 import Trigger from './components/Trigger';
 import type { Integration, IntegrationData } from './integrations/integration';
+import sentryDataCache from './integrations/sentry/data/sentryDataCache';
```

Comment:
That's not ideal, we should find another way to do this - we are adding something from the Sentry integration to the main skeleton app making Spotlight practically required to always use Sentry.

We should refactor this that Spotlight forwards all incoming events to all integrations. Integrations that also listen to the event do something then.

Reply:
Yes @HazAT , that makes sense.

In earlier commits, when we added clear events, I only implemented it for Sentry. However, this approach could potentially break the spotlight if the user hasn't added Sentry integration.

To address this, I've updated the logic by checking for the existence of the `processEvent` function for every integration. If the function is present, we clear the `integrationData` for that specific integration(which includes the events for every integration)
Additionally, for integrations like Sentry, which process events but have internal storage (like we have `sentryDataCache`), I've added a reset function in the integration. We will call this function while clearing events, allowing users to clear events in the integration using the button.

Approach Summary:
- Check for the existence of the processEvent function for each integration.
- If the function is present, clear the integration data (including events).
- For integrations with internal storage, implement a reset function in the integration.
- Call the reset function while clearing events to enable users to clear events using the button.